src/unix/fs.c

```diff
@@ -1240,9 +1241,28 @@ static ssize_t uv__fs_write(uv_fs_t* req) {
   return r;
 }
 
-static ssize_t uv__fs_copyfile(uv_fs_t* req) {
 #if defined(__APPLE__) && !TARGET_OS_IPHONE
-  /* On macOS, use the native copyfile(3). */
+
+static ssize_t uv__fs_clonefile_mac(uv_fs_t* req) {
+  static int has_clone_acl = 1;
+
+  /* Note: clonefile() does not set the group ID on the destination
+   * file correctly. */
+  if (!clonefile(req->path, req->new_path, has_clone_acl ? UV__FS_CLONE_ACL : 0))
+    return 0;  /* success */
+
+  if (errno == EINVAL) {
```

Comment:
I want to double-check: does clonefile() _always_ set errno on non-zero return value? If not, set `errno = 0` before the call.

Reply:
From the man page:

>  Upon successful completion, clonefile() returns 0. Otherwise, a value of -1 is returned and errno is set to indicate the error.